File: Data Structures & Algorithms/decode-ways/submission-6.cpp
class Solution {
public:
    int numDecodings(string s) {
        int dp[101];
        dp[s.size()]=1;
        for(int i=s.size()-1;i>=0;--i){
            if(s[i]=='0') dp[i]=0;
            else dp[i]= dp[i+1];
            if(i+1<s.size() and (s[i]=='1' or (s[i]=='2' and s[i+1]<'7'))) dp[i] += dp[i+2];
        }
        return dp[0];
    }
};

File: Data Structures & Algorithms/subsets-ii/submission-6.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        ans = set()
        nums.sort()
        for i in range(1<<len(nums)):
            temp_ans=[]
            for j in range(len(nums)):
                if i & (1<<j):
                    temp_ans.append(nums[j])
            ans.add(tuple(temp_ans))
        return list(list(_) for _ in ans)